### PR TITLE
[FEATURE] Since value set in expectation kwargs is list of strings, do not emit expect_column_values_to_be_in_set for datetime valued columns

### PR DESCRIPTION
**Pull request by alexsherstinsky**
### Scope
* Add condition to `categorical_columns_rule` of OnboardingDataAssistant to check for `parse_strings_as_datetimes` flag (indicates that datetime-valued column was detected), and if `True`, do not emit the `expect_column_values_to_be_in_set` expectation.
* Remove unary `~` operator from `condition` syntax in `DefaultExpectationConfigurationBuilder` -- it is incorrectly grouped as binary and is not used, and there is no unit test for it (it will be added properly later with a unit test).
* Update tests.

Please annotate your PR title to describe what the PR does, then give a brief bulleted description of your PR below. PR titles should begin with [BUGFIX], [FEATURE],  [DOCS], or [MAINTENANCE]. If a new feature introduces breaking changes for the Great Expectations API or configuration files, please also add [BREAKING]. You can read about the tags in our [contributor checklist](https://docs.greatexpectations.io/docs/contributing/contributing_checklist).

Changes proposed in this pull request:
- JIRA: GREAT-761/GREAT-1269
- 
-

After submitting your PR, CI checks will run and @cla-bot will check for your CLA signature.

For a PR with nontrivial changes, we review with both design-centric and code-centric lenses.

In a design review, we aim to ensure that the PR is consistent with our relationship to the open source community, with our software architecture and abstractions, and with our users' needs and expectations. That review often starts well before a PR, for example in github issues or slack, so please link to relevant conversations in notes below to help reviewers understand and approve your PR more quickly (e.g. `closes #123`).

Previous Design Review notes:
-
-
-

### Definition of Done
Please delete options that are not relevant.

- [x] My code follows the Great Expectations [style guide](https://docs.greatexpectations.io/docs/contributing/style_guides/code_style)
- [x] I have performed a [self-review](https://docs.greatexpectations.io/docs/contributing/contributing_checklist) of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added [unit tests](https://docs.greatexpectations.io/docs/contributing/contributing_test#writing-unit-and-integration-tests) where applicable and made sure that new and existing tests are passing.
- [x] I have run any local integration tests and made sure that nothing is broken.

Thank you for submitting!

### CODE REVIEW

File: great_expectations/rule_based_profiler/expectation_configuration_builder/default_expectation_configuration_builder.py [hunk 1]
```diff
@@ -33,7 +33,7 @@
 integer = Word(nums).setParseAction(lambda t: int(t[0]))
 var = Combine(Word("$" + alphas, alphanums + "_.") + ppOptional("[" + integer + "]"))
 comparison_operator = oneOf(">= <= != > < ==")
-binary_operator = oneOf("~ & |")
```
Comment (NathanFarmer) [inline great_expectations/rule_based_profiler/expectation_configuration_builder/default_expectation_configuration_builder.py:36]: Confirming this outstanding work was captured by GREAT-1270

File: great_expectations/rule_based_profiler/parameter_builder/value_set_multi_batch_parameter_builder.py [hunk 2]
```diff
@@ -125,15 +126,25 @@ def _build_parameters(
                 FULLY_QUALIFIED_PARAMETER_NAME_ATTRIBUTED_VALUE_KEY
             ]
         )
+        details: dict = parameter_node[FULLY_QUALIFIED_PARAMETER_NAME_METADATA_KEY]
+
+        unique_values: Set[Any] = _get_unique_values_from_nested_collection_of_sets(
+            collection=metric_values
+        )
+        unique_values_as_array: np.ndarray = np.asarray(unique_values)
+        if unique_values_as_array.ndim == 0:
```
Comment (NathanFarmer) [inline great_expectations/rule_based_profiler/parameter_builder/value_set_multi_batch_parameter_builder.py:135]: Confirming my understanding - if `unique_values` is a scalar, `np.asarray` will only return a scalar. This condition will turn the array into an `np.ndarray` of dimension 1.

**Review by NathanFarmer — APPROVED** [review]
Thanks for the synchronous review!

**Review by Shinnnyshinshin — APPROVED** [review]
two stamps! ✅✅

**Pull request merged**